astropy/cosmology/flrw.py

```diff
@@ -2262,7 +2257,7 @@ def __init__(self, H0, Om0, Ode0, w0=-1.0, Tcmb0=0.0*u.K, Neff=3.04,
                  m_nu=0.0*u.eV, Ob0=None, *, name=None, meta=None):
         super().__init__(H0=H0, Om0=Om0, Ode0=Ode0, Tcmb0=Tcmb0, Neff=Neff,
                          m_nu=m_nu, Ob0=Ob0, name=name, meta=meta)
-        self._w0 = float(w0)
+        self.w0 = float(w0)
```

Comment:
In principle, this could be done with `Parameter(..., fvalidate=float, ...)` - which perhaps suggests that it should be `validate` rather than `fvalidate`...

Reply:
Definitely yes to the first part.
I think ``fvalidate`` is good because its extending the ``fget/fset/fdel`` standard from property. Also ``Parameter.validate()`` calls ``fvalidate``, so it could be a little confusing to have an input match a function where they purposefully have different signatures.